### PR TITLE
Move to latest .NET8 pre-release version of NetAnalyzers package for dogfooding

### PR DESCRIPTION
**Pull request by mavasani**
**Recommended to be reviewed commit by commit**

This PR also fixes most of the new violations.

https://github.com/dotnet/roslyn-analyzers/issues/6379 tracks auditing certain suppressed violations so these are either fixed and suppressions removed or we decide that the suppressions are desirable.

### CODE REVIEW

**Comment by mavasani** [comment]
@Youssef1313 for review

File: eng/Versions.props [hunk 1]
```diff
@@ -47,7 +47,7 @@
     <MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>3.7.0</MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>
     <MicrosoftCodeAnalysisVersionForTests>4.4.0-2.22416.9</MicrosoftCodeAnalysisVersionForTests>
     <DogfoodAnalyzersVersion>3.3.4-beta1.22418.3</DogfoodAnalyzersVersion>
-    <DogfoodNetAnalyzersVersion>7.0.0</DogfoodNetAnalyzersVersion>
+    <DogfoodNetAnalyzersVersion>8.0.0-preview1.22621.6</DogfoodNetAnalyzersVersion>
```
Comment (Youssef1313) [inline eng/Versions.props:50]: So is 7.0.0 broken? If that's the case, I think it's important to fix that and release on NuGet?

File: src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/QualityGuidelines/CSharpAvoidMultipleEnumerationsAnalyzer.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@ namespace Microsoft.CodeAnalysis.CSharp.NetAnalyzers.Microsoft.CodeQuality.Analy
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     internal sealed partial class CSharpAvoidMultipleEnumerationsAnalyzer : AvoidMultipleEnumerations
     {
-        protected override bool IsExpressionOfForEachStatement(SyntaxNode node)
-            => node.Parent is ForEachStatementSyntax forEachStatementSyntax && forEachStatementSyntax.Expression.Equals(node);
+        protected override bool IsExpressionOfForEachStatement(SyntaxNode syntax)
+            => syntax.Parent is ForEachStatementSyntax forEachStatementSyntax && forEachStatementSyntax.Expression.Equals(syntax);
```
Comment (Youssef1313) [inline src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/QualityGuidelines/CSharpAvoidMultipleEnumerationsAnalyzer.cs:13]: Unrelated to the PR, but it looks like we have a false negative here. Guessing this should be `CommonForEachStatementSyntax`

File: eng/Versions.props (hunk 1 above)
Comment (mavasani) [reply eng/Versions.props:50]: Yes, AnalysisLevel/AnalysisMode regression is first fixed in 7.0.101. We definitely need to release the corresponding NuGet package on NuGet.org. Tagging @jmarolf

**Review by Youssef1313 — APPROVED** [review]
LGTM

**Comment by mavasani** [comment]
Thank you @Youssef1313

File: eng/Versions.props (hunk 1 above)
Comment (Youssef1313) [reply eng/Versions.props:50]: @mavasani The [`7.0.0` tag is pointing to e26a04c](https://github.com/dotnet/roslyn-analyzers/releases/tag/7.0.0), which is supposed to have the fix:

image

Is the tag pointing to incorrect commit?

Comment (mavasani) [reply eng/Versions.props:50]: I am guessing so

**Pull request merged**